src/connection.rs

```diff
@@ -427,44 +499,32 @@ impl ConnectionLike for Connection {
 /// }
 /// # }
 /// ```
-impl PubSub {
-    fn get_channel<T: ToRedisArgs>(&mut self, channel: &T) -> Vec<u8> {
-        let mut chan = vec![];
-        for item in channel.to_redis_args().iter() {
-            chan.extend(item.iter().cloned());
-        }
-        chan
+impl<'a> PubSub<'a> {
+    pub fn new(con: &'a mut Connection) -> Self {
```

Comment:
Do we need to expose this constructor? I'd favor to only have `as_pubsub` on the connection.

Reply:
Nope, this was a mistake. Nice catch.